src/Bicep.LangServer/Completions/BicepCompletionProvider.cs

```diff
@@ -909,6 +912,72 @@ private IEnumerable<CompletionItem> GetObjectBodyCompletions(TypeSymbol typeSymb
             }
         }
 
+        private IEnumerable<CompletionItem> GetDisableNextLineDiagnosticsDirectiveCompletion(BicepCompletionContext context)
+        {
+            if (context.Kind.HasFlag(BicepCompletionContextKind.DisableNextLineDiagnosticsDirectiveStart))
+            {
+                yield return CreateKeywordCompletion(LanguageConstants.DisableNextLineDiagnosticsKeyword, "Disable next line diagnostics directive", context.ReplacementRange);
+            }
+        }
+
+        private IEnumerable<CompletionItem> GetDisableNextLineDiagnosticsDirectiveCodesCompletion(SemanticModel model, BicepCompletionContext context)
+        {
+            if (context.Kind.HasFlag(BicepCompletionContextKind.DisableNextLineDiagnosticsCodes))
+            {
+                foreach (var diagnostic in GetDiagnosticCodes(context.ReplacementRange, model))
+                {
+                    yield return CreateKeywordCompletion(diagnostic.Code, diagnostic.Message, context.ReplacementRange);
+                }
+            }
+        }
+
+        private IEnumerable<IDiagnostic> GetDiagnosticCodes(Range range, SemanticModel model)
+        {
+            var lineStarts = model.SourceFile.LineStarts;
+            var position = GetPosition(range, lineStarts);
+            (var line, _) = TextCoordinateConverter.GetPosition(lineStarts, position);
+            var nextLineSpan = GetNextLineSpan(lineStarts, line, model.SourceFile.ProgramSyntax);
+
+            if (nextLineSpan is null)
+            {
+                return Enumerable.Empty<IDiagnostic>();
+            }
+
+            return model.GetAllDiagnostics()
+                .Where(diagnostic => nextLineSpan.ContainsInclusive(diagnostic.Span.Position) && nextLineSpan.ContainsInclusive(diagnostic.Span.GetEndPosition()))
+                .DistinctBy(x => x.Code);
```

Comment:
Need to ensure they're also suppressible here

Reply:
Resolving completion retriggers compilation, which should re calculate diagnostics. I added test to ensure that:
https://github.com/Azure/bicep/commit/42e587854a14be8ffbb5c50039517fb4858bba67

Let me know if I am missing something.

Reply:
Sorry, I should have explained better. In the SemanticModel, we check `IsSuppressible()` to ensure that you can't e.g. suppress core compiler errors. Here we should also check the same, as we don't want to offer to suppress a diagnostic which can't be suppressed.

Reply:
Ah got it. Thank you!

Updated here:
https://github.com/Azure/bicep/commit/b07d2e51ecf8b8e9b695cf524e3a1063e6a91e63